libraries/core_libs/network/src/tarcap/packets_handlers/dag_block_packet_handler.cpp

```diff
@@ -114,9 +114,41 @@ void DagBlockPacketHandler::sendBlock(dev::p2p::NodeID const &peer_id, taraxa::D
   }
 }
 
-void DagBlockPacketHandler::onNewBlockReceived(DagBlock &&block) {
+void DagBlockPacketHandler::onNewBlockReceived(DagBlock &&block, const std::shared_ptr<TaraxaPeer> &peer) {
   if (dag_blk_mgr_) {
-    dag_blk_mgr_->insertAndVerifyBlock(std::move(block));
+    const auto verified = dag_blk_mgr_->insertAndVerifyBlock(std::move(block));
+    switch (verified) {
+      case DagBlockManager::InsertAndVerifyBlockReturnType::InvalidBlock:
+      case DagBlockManager::InsertAndVerifyBlockReturnType::FailedVdfVerification:
+      case DagBlockManager::InsertAndVerifyBlockReturnType::NotEligible: {
+        std::ostringstream err_msg;
+        err_msg << "DagBlock" << block.getHash() << " failed verification with error code " << (uint32_t)verified;
+        throw MaliciousPeerException(err_msg.str());
+      }
+      case DagBlockManager::InsertAndVerifyBlockReturnType::BlockQueueOverflow:
+        LOG(log_wr_) << "DagBlock" << block.getHash() << " could not be inserted because of block queue overflow. Peer "
+                     << peer->getId() << " will be disconnected";
+        disconnect(peer->getId(), dev::p2p::UserReason);
+        break;
+      case DagBlockManager::InsertAndVerifyBlockReturnType::MissingTransaction:
+        if (peer->peer_dag_synced_) {
+          std::ostringstream err_msg;
+          err_msg << "DagBlock" << block.getHash() << " is missing a transaction while in a dag synced state";
+          throw MaliciousPeerException(err_msg.str());
+        }
+        break;
+      case DagBlockManager::InsertAndVerifyBlockReturnType::AheadBlock:
+      case DagBlockManager::InsertAndVerifyBlockReturnType::FutureBlock:
+        if (peer->peer_dag_synced_) {
+          LOG(log_wr_) << "DagBlock" << block.getHash() << " is an ahead/future block. Peer " << peer->getId()
+                       << " will be disconnected";
+          disconnect(peer->getId(), dev::p2p::UserReason);
+        }
+        break;
+      case DagBlockManager::InsertAndVerifyBlockReturnType::InsertedAndVerified:
```

Comment:
add break

Reply:
No, since case can handle more than one condition at the same time this way